demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp

```diff
@@ -112,6 +112,11 @@ struct Params {
         prepared_mat = params.prepared_mat;
         last_id.fetch_add(params.last_id);
     }
+    Params& operator=(const Params& params) {
```

Comment:
I think it's better to rely on struct initialization. You can remove constructor and this operator. And in `setup()` just do
```cpp
state = std::make_shared<BatchState::Params>({cv::Mat(), cv::Mat(
                                   std::vector<int>{1,
                                                    int(net_size[0]),
                                                    int(net_size[1]),
                                                    int(net_size[2]),
                                                    int(net_size[3])}
), 0});
```

Reply:
Unfortunately atomic<size_t> doesn't have copy constructor (deleted function). I am thinking about this now.